src/components/FileMenu/utils.js

```diff
@@ -45,11 +49,76 @@ export const appPathFor = (fileType, id, apiVersion) => {
     }
 }
 
-export const preparePayloadForSaveAs = ({ ...visualization }) => {
+export const preparePayloadForSaveAs = ({
+    visualization,
+    name,
+    description,
+}) => {
     delete visualization.id
     delete visualization.created
     delete visualization.createdBy
     delete visualization.user
 
+    visualization.name =
+        name ||
+        visualization.name ||
+        i18n.t('Untitled {{visualizationType}}, {{date}}', {
```

Comment:
Given the issues filtering with comma, should we remove that from the default name?

Reply:
The problem is not just the comma, apparently the space is used as token separator, so any `Untitled {{visualizationType}}` would match regardless of the date.